### PR TITLE
workflows/release-tasks: Setup FileCheck and not for release-lit

### LINKED ISSUE #64892: [lit] Tests seems to fail in our action
https://github.com/llvm/llvm-project/actions/runs/5936218533/job/16096191390

@tstellar

I ran the tests locally and it seems like locally it failed because `FileCheck` was not available, do we need to build `FileCheck` before we run the tests?

### PR DESCRIPTION
**Pull request by tuliom**
lit tests require commands FileCheck and not. They must be available in the PATH.

This also guarantees that python3-psutil is installed in order to enable more tests.

Fixes #64892.

### CODE REVIEW

**Comment by tru** [comment]
Thanks for the fix! I had that on my todo list. But I am worried that we use a fixed version from apt when I bet that lit tests will only be guaranteed to work with the file check from the same source. I think to be safe we need to build it.

**Comment by tru** [comment]
@jdenny-ornl @tstellar What do you think about this? How closely tied is lit with the current in-tree version of FileCheck? I would like to fix this before the next release so that we can run the lit tests in the CI.

**Comment by jdenny-ornl** [comment]
> @jdenny-ornl @tstellar What do you think about this? How closely tied is lit with the current in-tree version of FileCheck? I would like to fix this before the next release so that we can run the lit tests in the CI.

FileCheck doesn't grow new features frequently.  However, when it does, tests can use those features immediately.  I think it would be worthwhile to build the latest FileCheck.  @RoboTux @jh7370 might wish to comment as well.

**Comment by jh7370** [comment]
> > @jdenny-ornl @tstellar What do you think about this? How closely tied is lit with the current in-tree version of FileCheck? I would like to fix this before the next release so that we can run the lit tests in the CI.
> 
> FileCheck doesn't grow new features frequently. However, when it does, tests can use those features immediately. I think it would be worthwhile to build the latest FileCheck. @RoboTux @jh7370 might wish to comment as well.

I don't know enough about the context here to be able to provide clear feedback, but generally, I don't think it makes sense to use any LLVM tools other than the contemporary ones (i.e. the versions built from the same source as providing the tests) when running tests. Other tools that are considered part of "test supporting tools" include `count` and `split-file`, for example. It's likely that any new features added to these tools would be used immediately in tests too.

**Comment by RoboTux** [comment]
Likewise, I'm not sure what is the story behind this PR but since they are in the same tree I feel it should be fine to use the latest feature. It also sounds easier to just depend on the in-tree version with appropriate dependencies in CMake rather than detecting if it's available on the system, error out if not, and deal with bug report due to people using a too old version of lit (or test the version beforehand).

**Comment by tuliom** [comment]
I updated this PR in order to build both `FileCheck` and `not` as requested.

File: .github/workflows/release-tasks.yml
Comment (tru) [inline .github/workflows/release-tasks.yml:89]: I think we should maybe use the `setup-cpp` action that installs clang from our apt.llvm.org instead and we can pin the version, so it doesn't change under us randomly and then we get Ninja as well.

File: .github/workflows/release-tasks.yml
Comment (tru) [inline .github/workflows/release-tasks.yml:94]: Using ninja is going to be faster.

File: .github/workflows/release-tasks.yml
Comment (tru) [inline .github/workflows/release-tasks.yml:101]: is this needed? I think if you invoke lit from the build dir it will pick up filecheck from the same dir.

File: .github/workflows/release-tasks.yml
Comment (tru) [reply .github/workflows/release-tasks.yml:89]:
```
  - name: Setup Cpp
    uses: aminya/setup-cpp@v1
    with:
      compiler: llvm-16.0.6
      cmake: true
      ninja: true
```

**Review by tru — COMMENTED** [review]
Some ideas for improvements. But in general nice work!

File: .github/workflows/release-tasks.yml
Comment (tuliom) [reply .github/workflows/release-tasks.yml]: I've just pushed a fix for this.

File: .github/workflows/release-tasks.yml
Comment (tuliom) [reply .github/workflows/release-tasks.yml]: Done.

File: .github/workflows/release-tasks.yml
Comment (tuliom) [reply .github/workflows/release-tasks.yml]: Done.

Comment (tuliom) [reply .github/workflows/release-tasks.yml]: Considering RoboTux's suggestion, I'm closing this as resolved.

File: .github/workflows/release-tasks.yml
Comment (jdenny-ornl) [inline .github/workflows/release-tasks.yml:102]: To make debugging any failed tests easier, would you please consider adding the following to the environment when running check-lit?

```
export FILECHECK_OPTS='-dump-input-filter=all -vv -color'
```

Comment (jdenny-ornl) [reply .github/workflows/release-tasks.yml:102]: On second thought, I realize I don't know that I understand the context.  If this is run manually and can easily be re-run with different environment variables, then the person running it should decide whether to make this change.  Otherwise, defaulting to verbosity would be helpful for whoever has to understand test failures.

Comment (tuliom) [reply .github/workflows/release-tasks.yml:103]: It doesn't hurt to have extra verbosity.
Running this is not complex, but does require a specific setup.
I added included this suggestion in my last commit.

Comment (jdenny-ornl) [reply .github/workflows/release-tasks.yml:103]: Thanks!

**Comment by tru** [comment]
Thanks for your work on this! I think we should merge this, but probably also spin it out to its own action to run on the release branch PRs since this current workflow only runs when we make a new release. And at that point it's almost "too late". We ideally would use this feedback earlier in the release to fix issues.

**Review by tru — APPROVED** [review]

**Review by RoboTux — APPROVED** [review]
LGTM

**Comment by tru** [comment]
@tuliom planning on merging this - or do you need someone with access to do that?

**Comment by tuliom** [comment]
> @tuliom planning on merging this - or do you need someone with access to do that?

@tru I'm doing this now. Thanks!

**Comment by tuliom** [comment]
I've just updated the description in order to remove a paragraph that didn't make sense anymore after the reviews and squashed all commits.
I will merge it after the tests get executed and the merge button gets green again.

**Pull request merged**
